### ISSUE TITLE
zero width space in deserialized jsv

### ISSUE THREAD

**Issue by mobileBlox** [ServiceStack/Issues#723]
Hi,

recently we've updated all ServiceStack libs and also updated to .NET 4.8. We've found an issue with the jsv deserializer in the latest version (5.8.0). Actually empty/null strings have a zero width space in the result.

```csharp
var value = "{Number:10330,CountryKey:DE,FederalStateKey:\"\",Salutation:,City:''}";
var map = (Dictionary<string, object>)value.FromJsv<object>();

foreach (var entry in map)
{
	System.Diagnostics.Debug.Write(entry.Key);
	int length = ((string)entry.Value).Length;
	System.Diagnostics.Debug.Write(", Length: " + length);

	if (length > 0)
	{
		System.Diagnostics.Debug.Write(", Value: " + (entry.Value));
	}

	System.Diagnostics.Debug.WriteLine("");
}
```

Output:
Number, Length: 5, Value: 10330
CountryKey, Length: 2, Value: DE
FederalStateKey, Length: 1, Value: ​
Salutation, Length: 0
City, Length: 2, Value: ''

The **FederalStateKey should not have any value**. But it contains a zero width space from the following output of the Visual Studio Immediate Window:
((string)map["FederalStateKey"])[0]
8203 '​'

EDIT: sorry, the code tag does not work properly for this code snippet

**Comment by mythz** [comment]
As deserializing an object doesn't return a `Dictionary<string, object>` by default I'm assuming you're using non-standard configuration like `JsConfig.ConvertObjectTypesIntoStringDictionary`? 

In future please include any non standard configuration required to run posted samples so the issue can be properly repro'd & resolved. 

But I was able to repro it with `JsConfig.ConvertObjectTypesIntoStringDictionary`, which should now be resolved in [this commit](https://github.com/ServiceStack/ServiceStack.Text/commit/71529920e78fed36f8ac0bda6ff277232da9e4bf).

This change is available from the latest v5.8.1 that's now [available on MyGet](https://docs.servicestack.net/myget).

**Comment by mobileBlox** [comment]
Thank you very much!

You're right, I've forgot to include the JsConfig.ConvertObjectTypesIntoStringDictionary = true line, sorry.
